discovery-provider/src/queries/query_helpers.py

```diff
@@ -787,10 +698,10 @@ def populate_playlist_metadata(session, playlist_ids, playlists, repost_types, s
 
     for playlist in playlists:
         playlist_id = playlist["playlist_id"]
-        playlist[response_name_constants.repost_count] = playlist_repost_counts.get(
-            playlist_id, 0)
-        playlist[response_name_constants.save_count] = playlist_save_counts.get(
-            playlist_id, 0)
+        playlist[response_name_constants.repost_count] = count_dict.get(
+            playlist_id, {}).get(response_name_constants.repost_count, 0)
+        playlist[response_name_constants.save_count] = count_dict.get(
+            playlist_id, {}).get(response_name_constants.save_count, 0)
```

Comment:
Just putting this comment here since GH won't let me put it down below - what about the `get_repost_counts_query` function below here? Do we use that somewhere / would it benefit from the matview?

And `create_save_repost_count_subquery` waaay down there? There are a few functions at the bottom of this file that may be candidates for matview.

Reply:
Yeah, I didn't want to touch `get_repost_counts_query` because it's really dynamic in terms of allowing filter_ids, repost_type, max_block_number, and filter by user_id & repost_item_id.

We could add a short circuit to use the mat view in a specific case, but I think it may be a little messy, so I wanted to avoid it for now.

Reply:
Yeah I was thinking short circuit. Let's just flag it for later, maybe linear or something?